dms/models/storage.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2017-2019 MuK IT GmbH.
+# Copyright 2020 Creu Blanca
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import AccessError
+
+_logger = logging.getLogger(__name__)
+
+
+class Storage(models.Model):
+
+    _name = "dms.storage"
+    _description = "Storage"
+
+    # ----------------------------------------------------------
+    # Database
+    # ----------------------------------------------------------
+
+    name = fields.Char(string="Name", required=True)
+
+    save_type = fields.Selection(
+        selection=[("database", _("Database")), ("file", _("Filestore"))],
+        string="Save Type",
+        default="database",
+        required=True,
+        help="""The save type is used to determine how a file is saved by the
+        system. If you change this setting, you can migrate existing files
+        manually by triggering the action.""",
+    )
+
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        default=lambda self: self.env.user.company_id,
+        help="If set, directories and files will only be available for "
+        "the selected company.",
+    )
+
+    is_hidden = fields.Boolean(
+        string="Storage is Hidden",
+        default=False,
+        help="Indicates if directories and files are hidden by default.",
+    )
+
+    root_directory_ids = fields.One2many(
+        comodel_name="dms.directory",
+        inverse_name="root_storage_id",
+        string="Root Directories",
+        auto_join=False,
+        readonly=False,
+        copy=False,
+    )
+
+    storage_directory_ids = fields.One2many(
+        comodel_name="dms.directory",
+        inverse_name="storage_id",
+        string="Directories",
+        auto_join=False,
+        readonly=True,
+        copy=False,
+    )
+
+    storage_file_ids = fields.One2many(
+        comodel_name="dms.file",
+        inverse_name="storage_id",
+        string="Files",
+        auto_join=False,
+        readonly=True,
+        copy=False,
+    )
+
+    count_storage_directories = fields.Integer(
+        compute="_compute_count_storage_directories", string="Count Directories"
+    )
+
+    count_storage_files = fields.Integer(
+        compute="_compute_count_storage_files", string="Count Files"
+    )
+
+    # ----------------------------------------------------------
+    # Actions
+    # ----------------------------------------------------------
+
+    def action_storage_migrate(self):
```

Comment:
Isn't make sense to auto-execute this when `save_type` is changed and remove the manual button?

Reply:
Yes and No, I mean, if you have a small set of elements, it could be done this way, otherwise, you might need to do it on small batches. :thinking: